WooCommerce/Classes/ViewModels/MainTabViewModel.swift

```diff
@@ -67,23 +89,15 @@ final class MainTabViewModel {
     /// Loads the the hub Menu tab badge and listens to any change to update it
     ///
     func loadHubMenuTabBadge() {
-        updateHubMenuTabBadge()
-
-        NotificationCenter.default.addObserver(self,
-                                               selector: #selector(updateHubMenuTabBadge),
-                                               name: .reviewsBadgeReloadRequired,
-                                               object: nil)
-    }
+        synchronizeShouldShowBadgeOnHubMenuTabLogic()
 
-    @objc func updateHubMenuTabBadge() {
-        guard let siteID = storesManager.sessionManager.defaultStoreID else {
-            return
-        }
+        listenToReviewsBadgeReloadRequired()
+        retrieveShouldShowReviewsBadgeOnHubMenuTabValue()
 
-        let action = NotificationCountAction.load(siteID: siteID, type: .kind(.comment)) { [weak self] count in
-            self?.showMenuBadge?(count > 0, .primary)
+        if featureFlagService.isFeatureFlagEnabled(.paymentsHubMenuSection) {
+            listenToNewFeatureBadgeReloadRequired()
+            retrieveShouldShowNewFeatureBadgeOnHubMenuTabValue()
```

Comment:
Nice! Much more clear now that everything is into their own method 💯

Reply:
Thanks!